runtime/src/bank_forks.rs

```diff
@@ -201,11 +219,17 @@ impl BankForks {
         bank
     }
 
-    pub fn insert_from_ledger(&mut self, bank: Bank) -> Arc<Bank> {
+    pub fn insert_from_ledger(&mut self, bank: Bank) -> BankWithScheduler {
         self.highest_slot_at_startup = std::cmp::max(self.highest_slot_at_startup, bank.slot());
         self.insert(bank)
     }
 
+    // gating this under #[cfg(feature = "dev-context-only-utils")] isn't easy due to
+    // solana-program-test's usage...
+    pub fn insert_without_scheduler(&mut self, bank: Bank) -> Arc<Bank> {
```

Comment:
The return type is different so we only get the `Arc<Bank>` back, but since `BankWithScheduler` implements `Deref` w/ `Target=&Arc<Bank>`, it doesn't seem that makes any of the calling code different (w/ the exception of cloning?)

Reply:
> Can you elaborate on what situations we would want to insert without a scheduler?

well, there's no concrete situations... `insert_without_scheduler()` was defined as a helper to omit `.clone_without_scheduler()` at call sites. but seems i went too far. just removed fec59c9196a1b6d52a739f47af384f5f6cd7210f

> Looking at the main PR's code for insert, we should be inserting without a scheduler so long as the BankForks does not have a scheduler pool, which I believe will be the default.

yeah, this is correct.

> The return type is different so we only get the Arc<Bank> back, but since BankWithScheduler implements Deref w/ Target=&Arc<Bank>, it doesn't seem that makes any of the calling code different (w/ the exception of cloning?)

again, yes, this observation is correct as well.

some existing api takes `Arc<Bank>` ownership. so, we need to convert back to `Arc<Bank>` from `BankWithScheduler` in some way here and there.

Reply:
> some existing api takes Arc<Bank> ownership.

:thinking: Yeah, I can see that now from the commit diff that we need to have the `Arc<Bank>` in several call locations.
I think the fec59c9 commit is the right direction, in that we shouldn't be adding a separate insertion path for that use-case.